src/idl_parser.cpp

```diff
@@ -970,6 +970,14 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
   FieldDef *field;
   ECHECK(AddField(struct_def, name, type, &field));
 
+  if (typefield) {
```

Comment:
IMO, It can be moved 10 line upper and remove this if, I mean It can be in `if (type.base_type == BASE_TYPE_UNION)` block

Reply:
not possible as field is assigned on line 971